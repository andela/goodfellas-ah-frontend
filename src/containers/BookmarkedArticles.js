import React, { Component } from 'react';
import { connect } from 'react-redux';
import { userPlaceholderImage } from '../mixin';
import { Loader as Loading } from '../components/shared/Loading';
import { fetchProfile } from '../actions/profileActions';
import { getBookmarkedArticles } from '../actions/articleActions';
import BookmarkedArticleList from '../components/article/BookmarkedArticleList';

// eslint-disable-next-line react/prefer-stateless-function
class BookmarkedArticles extends Component {
  componentWillMount = () => {
    const {
      user: { userId },
      fetchProfile: fetchUserProfile,
      getBookmarkedArticles: fetchBookmarkedArticles,
    } = this.props;
    fetchUserProfile(userId);
    fetchBookmarkedArticles();
  }

  render() {
    const { profileStore, bookmarked } = this.props;
    const fullName = `${profileStore.user.firstname} ${profileStore.user.lastname}`;
    if (profileStore.loading && !bookmarked) return <Loading />;
    return (
      <div>
        <div className="profile-details">
          <img alt="profile" className="profile-image" src={profileStore.profile.image || userPlaceholderImage} />
          <h3 id="user-name" className="username">{fullName}</h3>
        </div>
        <h3 className="profile-title">Bookmarks</h3>
        <BookmarkedArticleList articles={bookmarked} />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  profileStore: state.profile,
  user: state.auth.ownProfile,
  bookmarked: state.articles.bookmarkedArticles,
});

export default connect(mapStateToProps, { fetchProfile, getBookmarkedArticles })(BookmarkedArticles);
