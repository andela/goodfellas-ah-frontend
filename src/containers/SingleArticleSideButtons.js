import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import icons from '../assets/icons.svg';
import { react as reactAction } from '../actions/articleActions';

export class SideButtons extends Component {
  state = {
    extraButtons: false,
    showSideButton: true,
  }

  authorizedButtons = () => {
    const { userId, article } = this.props;
    if (userId === article.authorId) {
      return (
        <div>
          <svg id="delete" className="icon" title="Delete this article"><use xlinkHref={`${icons}#delete-icon`} /></svg>
          <svg id="edit" title="Edit this article"><Link className="link" to={`/articles/edit/${article.slug}`}><use xlinkHref={`${icons}#edit`} /></Link></svg>
        </div>
      );
    }
  }

  showExtraButtons = () => {
    this.setState((prevState) => ({ extraButtons: !prevState.extraButtons }));
  }

  toggleSideButton = () => {
    this.setState((prevState) => ({ showSideButton: !prevState.showSideButton }));
  }

  AddExtraButtons = (className) => {
    const { article } = this.props;
    return (
      <div className={`extra-buttons ${className}`}>
        {this.authorizedButtons()}
        <svg id="flag" title="Report this article"><use xlinkHref={`${icons}#flag`} /></svg>
        <svg id="dislike" className={article.myReactions[0] && article.myReactions[0].reaction === -1 ? 'active' : ''} onClick={() => this.addReaction(-1)} title="Dislike this article"><use xlinkHref={`${icons}#dislike`} /></svg>
      </div>
    );
  };

  addReaction = async (reaction) => {
    const { react, article } = this.props;
    await react(article.slug, reaction);
  }

  render() {
    const { extraButtons, showSideButton } = this.state;
    const { article } = this.props;
    return (
      <Fragment>
        <span className="small-button" onClick={this.toggleSideButton}>click</span>
        <div className={`side-button ${showSideButton ? 'active' : ''}`}>
          <svg id="star" title="Like this article"><use xlinkHref={`${icons}#star`} /></svg>
          <svg id="bookmark" title="Report this article"><use xlinkHref={`${icons}#bookmark`} /></svg>
          <svg id="share" title="Share this article"><use xlinkHref={`${icons}#share`} /></svg>
          <svg id="more" onClick={this.showExtraButtons} title="Show more buttons"><use xlinkHref={`${icons}#more`} /></svg>
          {this.AddExtraButtons(extraButtons ? 'active' : '')}
          <div id="like-count"><p>{article.reactionCount.likes}</p></div>
          <span id="love" onClick={() => this.addReaction()} className={`centralizer ${article.myReactions[0] && article.myReactions[0].reaction === 1 ? 'liked' : ''}`}>
            <svg className="icon"><use xlinkHref={`${icons}#heart`} /></svg>
          </span>
        </div>
      </Fragment>
    );
  }
}


const mapStateToProps = (state) => ({
  article: state.articles.article,
});

export default connect(
  mapStateToProps,
  { react: reactAction },
)(SideButtons);
