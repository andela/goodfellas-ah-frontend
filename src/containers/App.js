import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import '../styles/styles.scss';
import authenticate from './hoc/authenticate';
import LandingPage from '../views/LandingPage';
import Signin from '../views/SigninPage';
import Signup from '../views/SignupPage';
import SingleArticle from '../views/SingleArticlePage';
import EditProfile from '../views/ProfileEditPage';
import Profile from '../views/ProfilePage';
import SocialAuthPage from '../views/SocialAuthPage';
import ForgotPasswordPage from '../views/ForgotPasswordPage';
import ResetPasswordPage from '../views/ResetPasswordPage';
import SearchArticles from '../views/SearchArticles';
import NotificationPage from '../views/NotificationPage';
import GetAllArticles from '../views/AllArticles';
import Tags from '../components/article/Tags';
import CreateArticle from '../views/CreateArticles';
import UpdateArticle from '../views/UpdateArticle';
import BookmarkedArticles from '../views/BookmarkedArticlesPage';

const User = () => (
  <Switch>
    <Route exact path="/user/profile" component={authenticate(Profile)} />
    <Route exact path="/user/profile/edit" component={authenticate(EditProfile)} />
    <Route exact path="/user/profile/:userId" component={Profile} />
    <Route path="/user/notifications" component={NotificationPage} />
  </Switch>
);

const Auth = () => (
  <div>
    <Route path="/auth/signin" render={(props) => <Signin formtype="signin" {...props} />} />
    <Route path="/auth/signup" render={(props) => <Signup formtype="signup" {...props} />} />
    <Route path="/auth/social" component={SocialAuthPage} />
  </div>
);

const Articles = () => (
  <div>
    <Switch>
      <Route path="/articles/tags/create" component={Tags} />
      <Route path="/articles/search" component={SearchArticles} />
      <Route path="/articles/create" component={authenticate(CreateArticle)} />
      <Route path="/articles/bookmarked" component={authenticate(BookmarkedArticles)} />
      <Route path="/articles/home" component={GetAllArticles} />
      <Route path="/articles/edit/:slug" component={authenticate(UpdateArticle)} />
      <Route path="/articles/:slug" component={SingleArticle} />
    </Switch>
  </div>
);

const App = () => (
  <BrowserRouter>
    <div>
      <Route exact path="/" component={LandingPage} />
      <Route path="/auth" component={Auth} />
      <Route path="/articles" component={Articles} />
      <Route path="/user" component={User} />
      <Route path="/forgotpassword" component={ForgotPasswordPage} />
      <Route path="/resetpassword" component={ResetPasswordPage} />
    </div>
  </BrowserRouter>
);

export default App;
