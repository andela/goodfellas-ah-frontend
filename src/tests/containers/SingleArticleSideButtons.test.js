import React from 'react';
import { shallow } from 'enzyme';
import { SideButtons } from '../../containers/SingleArticleSideButtons';
import { articleResponse } from '../mock/articleData';

let wrapped;
const props = {
  article: articleResponse.data.data,
  react: jest.fn(),
};

beforeEach(() => {
  wrapped = shallow(<SideButtons {...props} />);
});


describe('SideButtons UI', () => {
  it('view should render as expected', () => {
    wrapped.find('#more').simulate('click');
    expect(wrapped).toMatchSnapshot();
  });
  it('it should react when like is clicked', () => {
    wrapped.find('#love').simulate('click');
    expect(props.react).toHaveBeenCalledWith('a-new-article', undefined);
  });
  it('it should react when dislike is clicked', () => {
    wrapped.find('#dislike').simulate('click');
    expect(props.react).toHaveBeenCalledWith('a-new-article', -1);
  });
});
