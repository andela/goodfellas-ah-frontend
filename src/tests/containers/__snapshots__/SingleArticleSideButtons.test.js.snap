// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideButtons UI view should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideButtons
    article={
      Object {
        "body": "content of the new article",
        "myReactions": Array [
          Object {
            "reaction": 1,
          },
        ],
        "reactionCount": Object {
          "dislikes": 5,
          "likes": 5,
        },
        "slug": "a-new-article",
        "title": "A new Article",
      }
    }
    react={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span
          className="small-button"
          onClick={[Function]}
        >
          click
        </span>,
        <div
          className="side-button active"
        >
          <svg
            id="star"
            title="Like this article"
          >
            <use
              xlinkHref="icons.svg#star"
            />
          </svg>
          <svg
            id="bookmark"
            title="Report this article"
          >
            <use
              xlinkHref="icons.svg#bookmark"
            />
          </svg>
          <svg
            id="share"
            title="Share this article"
          >
            <use
              xlinkHref="icons.svg#share"
            />
          </svg>
          <svg
            id="more"
            onClick={[Function]}
            title="Show more buttons"
          >
            <use
              xlinkHref="icons.svg#more"
            />
          </svg>
          <div
            className="extra-buttons active"
          >
            <div>
              <svg
                className="icon"
                id="delete"
                title="Delete this article"
              >
                <use
                  xlinkHref="icons.svg#delete-icon"
                />
              </svg>
              <svg
                id="edit"
                title="Edit this article"
              >
                <Link
                  className="link"
                  replace={false}
                  to="/articles/edit/a-new-article"
                >
                  <use
                    xlinkHref="icons.svg#edit"
                  />
                </Link>
              </svg>
            </div>
            <svg
              id="flag"
              title="Report this article"
            >
              <use
                xlinkHref="icons.svg#flag"
              />
            </svg>
            <svg
              className=""
              id="dislike"
              onClick={[Function]}
              title="Dislike this article"
            >
              <use
                xlinkHref="icons.svg#dislike"
              />
            </svg>
          </div>
          <div
            id="like-count"
          >
            <p>
              5
            </p>
          </div>
          <span
            className="centralizer liked"
            id="love"
            onClick={[Function]}
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#heart"
              />
            </svg>
          </span>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "click",
          "className": "small-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "click",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <svg
              id="star"
              title="Like this article"
            >
              <use
                xlinkHref="icons.svg#star"
              />
            </svg>,
            <svg
              id="bookmark"
              title="Report this article"
            >
              <use
                xlinkHref="icons.svg#bookmark"
              />
            </svg>,
            <svg
              id="share"
              title="Share this article"
            >
              <use
                xlinkHref="icons.svg#share"
              />
            </svg>,
            <svg
              id="more"
              onClick={[Function]}
              title="Show more buttons"
            >
              <use
                xlinkHref="icons.svg#more"
              />
            </svg>,
            <div
              className="extra-buttons active"
            >
              <div>
                <svg
                  className="icon"
                  id="delete"
                  title="Delete this article"
                >
                  <use
                    xlinkHref="icons.svg#delete-icon"
                  />
                </svg>
                <svg
                  id="edit"
                  title="Edit this article"
                >
                  <Link
                    className="link"
                    replace={false}
                    to="/articles/edit/a-new-article"
                  >
                    <use
                      xlinkHref="icons.svg#edit"
                    />
                  </Link>
                </svg>
              </div>
              <svg
                id="flag"
                title="Report this article"
              >
                <use
                  xlinkHref="icons.svg#flag"
                />
              </svg>
              <svg
                className=""
                id="dislike"
                onClick={[Function]}
                title="Dislike this article"
              >
                <use
                  xlinkHref="icons.svg#dislike"
                />
              </svg>
            </div>,
            <div
              id="like-count"
            >
              <p>
                5
              </p>
            </div>,
            <span
              className="centralizer liked"
              id="love"
              onClick={[Function]}
            >
              <svg
                className="icon"
              >
                <use
                  xlinkHref="icons.svg#heart"
                />
              </svg>
            </span>,
          ],
          "className": "side-button active",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#star"
              />,
              "id": "star",
              "title": "Like this article",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#star",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#bookmark"
              />,
              "id": "bookmark",
              "title": "Report this article",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#bookmark",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#share"
              />,
              "id": "share",
              "title": "Share this article",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#share",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#more"
              />,
              "id": "more",
              "onClick": [Function],
              "title": "Show more buttons",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#more",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <svg
                    className="icon"
                    id="delete"
                    title="Delete this article"
                  >
                    <use
                      xlinkHref="icons.svg#delete-icon"
                    />
                  </svg>
                  <svg
                    id="edit"
                    title="Edit this article"
                  >
                    <Link
                      className="link"
                      replace={false}
                      to="/articles/edit/a-new-article"
                    >
                      <use
                        xlinkHref="icons.svg#edit"
                      />
                    </Link>
                  </svg>
                </div>,
                <svg
                  id="flag"
                  title="Report this article"
                >
                  <use
                    xlinkHref="icons.svg#flag"
                  />
                </svg>,
                <svg
                  className=""
                  id="dislike"
                  onClick={[Function]}
                  title="Dislike this article"
                >
                  <use
                    xlinkHref="icons.svg#dislike"
                  />
                </svg>,
              ],
              "className": "extra-buttons active",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <svg
                      className="icon"
                      id="delete"
                      title="Delete this article"
                    >
                      <use
                        xlinkHref="icons.svg#delete-icon"
                      />
                    </svg>,
                    <svg
                      id="edit"
                      title="Edit this article"
                    >
                      <Link
                        className="link"
                        replace={false}
                        to="/articles/edit/a-new-article"
                      >
                        <use
                          xlinkHref="icons.svg#edit"
                        />
                      </Link>
                    </svg>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <use
                        xlinkHref="icons.svg#delete-icon"
                      />,
                      "className": "icon",
                      "id": "delete",
                      "title": "Delete this article",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "xlinkHref": "icons.svg#delete-icon",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "use",
                    },
                    "type": "svg",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="link"
                        replace={false}
                        to="/articles/edit/a-new-article"
                      >
                        <use
                          xlinkHref="icons.svg#edit"
                        />
                      </Link>,
                      "id": "edit",
                      "title": "Edit this article",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": <use
                          xlinkHref="icons.svg#edit"
                        />,
                        "className": "link",
                        "replace": false,
                        "to": "/articles/edit/a-new-article",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "xlinkHref": "icons.svg#edit",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "use",
                      },
                      "type": [Function],
                    },
                    "type": "svg",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <use
                    xlinkHref="icons.svg#flag"
                  />,
                  "id": "flag",
                  "title": "Report this article",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "xlinkHref": "icons.svg#flag",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "use",
                },
                "type": "svg",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <use
                    xlinkHref="icons.svg#dislike"
                  />,
                  "className": "",
                  "id": "dislike",
                  "onClick": [Function],
                  "title": "Dislike this article",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "xlinkHref": "icons.svg#dislike",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "use",
                },
                "type": "svg",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                5
              </p>,
              "id": "like-count",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
              },
              "ref": null,
              "rendered": 5,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                className="icon"
              >
                <use
                  xlinkHref="icons.svg#heart"
                />
              </svg>,
              "className": "centralizer liked",
              "id": "love",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <use
                  xlinkHref="icons.svg#heart"
                />,
                "className": "icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "xlinkHref": "icons.svg#heart",
                },
                "ref": null,
                "rendered": null,
                "type": "use",
              },
              "type": "svg",
            },
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span
            className="small-button"
            onClick={[Function]}
          >
            click
          </span>,
          <div
            className="side-button active"
          >
            <svg
              id="star"
              title="Like this article"
            >
              <use
                xlinkHref="icons.svg#star"
              />
            </svg>
            <svg
              id="bookmark"
              title="Report this article"
            >
              <use
                xlinkHref="icons.svg#bookmark"
              />
            </svg>
            <svg
              id="share"
              title="Share this article"
            >
              <use
                xlinkHref="icons.svg#share"
              />
            </svg>
            <svg
              id="more"
              onClick={[Function]}
              title="Show more buttons"
            >
              <use
                xlinkHref="icons.svg#more"
              />
            </svg>
            <div
              className="extra-buttons active"
            >
              <div>
                <svg
                  className="icon"
                  id="delete"
                  title="Delete this article"
                >
                  <use
                    xlinkHref="icons.svg#delete-icon"
                  />
                </svg>
                <svg
                  id="edit"
                  title="Edit this article"
                >
                  <Link
                    className="link"
                    replace={false}
                    to="/articles/edit/a-new-article"
                  >
                    <use
                      xlinkHref="icons.svg#edit"
                    />
                  </Link>
                </svg>
              </div>
              <svg
                id="flag"
                title="Report this article"
              >
                <use
                  xlinkHref="icons.svg#flag"
                />
              </svg>
              <svg
                className=""
                id="dislike"
                onClick={[Function]}
                title="Dislike this article"
              >
                <use
                  xlinkHref="icons.svg#dislike"
                />
              </svg>
            </div>
            <div
              id="like-count"
            >
              <p>
                5
              </p>
            </div>
            <span
              className="centralizer liked"
              id="love"
              onClick={[Function]}
            >
              <svg
                className="icon"
              >
                <use
                  xlinkHref="icons.svg#heart"
                />
              </svg>
            </span>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "click",
            "className": "small-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "click",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <svg
                id="star"
                title="Like this article"
              >
                <use
                  xlinkHref="icons.svg#star"
                />
              </svg>,
              <svg
                id="bookmark"
                title="Report this article"
              >
                <use
                  xlinkHref="icons.svg#bookmark"
                />
              </svg>,
              <svg
                id="share"
                title="Share this article"
              >
                <use
                  xlinkHref="icons.svg#share"
                />
              </svg>,
              <svg
                id="more"
                onClick={[Function]}
                title="Show more buttons"
              >
                <use
                  xlinkHref="icons.svg#more"
                />
              </svg>,
              <div
                className="extra-buttons active"
              >
                <div>
                  <svg
                    className="icon"
                    id="delete"
                    title="Delete this article"
                  >
                    <use
                      xlinkHref="icons.svg#delete-icon"
                    />
                  </svg>
                  <svg
                    id="edit"
                    title="Edit this article"
                  >
                    <Link
                      className="link"
                      replace={false}
                      to="/articles/edit/a-new-article"
                    >
                      <use
                        xlinkHref="icons.svg#edit"
                      />
                    </Link>
                  </svg>
                </div>
                <svg
                  id="flag"
                  title="Report this article"
                >
                  <use
                    xlinkHref="icons.svg#flag"
                  />
                </svg>
                <svg
                  className=""
                  id="dislike"
                  onClick={[Function]}
                  title="Dislike this article"
                >
                  <use
                    xlinkHref="icons.svg#dislike"
                  />
                </svg>
              </div>,
              <div
                id="like-count"
              >
                <p>
                  5
                </p>
              </div>,
              <span
                className="centralizer liked"
                id="love"
                onClick={[Function]}
              >
                <svg
                  className="icon"
                >
                  <use
                    xlinkHref="icons.svg#heart"
                  />
                </svg>
              </span>,
            ],
            "className": "side-button active",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <use
                  xlinkHref="icons.svg#star"
                />,
                "id": "star",
                "title": "Like this article",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "xlinkHref": "icons.svg#star",
                },
                "ref": null,
                "rendered": null,
                "type": "use",
              },
              "type": "svg",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <use
                  xlinkHref="icons.svg#bookmark"
                />,
                "id": "bookmark",
                "title": "Report this article",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "xlinkHref": "icons.svg#bookmark",
                },
                "ref": null,
                "rendered": null,
                "type": "use",
              },
              "type": "svg",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <use
                  xlinkHref="icons.svg#share"
                />,
                "id": "share",
                "title": "Share this article",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "xlinkHref": "icons.svg#share",
                },
                "ref": null,
                "rendered": null,
                "type": "use",
              },
              "type": "svg",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <use
                  xlinkHref="icons.svg#more"
                />,
                "id": "more",
                "onClick": [Function],
                "title": "Show more buttons",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "xlinkHref": "icons.svg#more",
                },
                "ref": null,
                "rendered": null,
                "type": "use",
              },
              "type": "svg",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <svg
                      className="icon"
                      id="delete"
                      title="Delete this article"
                    >
                      <use
                        xlinkHref="icons.svg#delete-icon"
                      />
                    </svg>
                    <svg
                      id="edit"
                      title="Edit this article"
                    >
                      <Link
                        className="link"
                        replace={false}
                        to="/articles/edit/a-new-article"
                      >
                        <use
                          xlinkHref="icons.svg#edit"
                        />
                      </Link>
                    </svg>
                  </div>,
                  <svg
                    id="flag"
                    title="Report this article"
                  >
                    <use
                      xlinkHref="icons.svg#flag"
                    />
                  </svg>,
                  <svg
                    className=""
                    id="dislike"
                    onClick={[Function]}
                    title="Dislike this article"
                  >
                    <use
                      xlinkHref="icons.svg#dislike"
                    />
                  </svg>,
                ],
                "className": "extra-buttons active",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <svg
                        className="icon"
                        id="delete"
                        title="Delete this article"
                      >
                        <use
                          xlinkHref="icons.svg#delete-icon"
                        />
                      </svg>,
                      <svg
                        id="edit"
                        title="Edit this article"
                      >
                        <Link
                          className="link"
                          replace={false}
                          to="/articles/edit/a-new-article"
                        >
                          <use
                            xlinkHref="icons.svg#edit"
                          />
                        </Link>
                      </svg>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <use
                          xlinkHref="icons.svg#delete-icon"
                        />,
                        "className": "icon",
                        "id": "delete",
                        "title": "Delete this article",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "xlinkHref": "icons.svg#delete-icon",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "use",
                      },
                      "type": "svg",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          className="link"
                          replace={false}
                          to="/articles/edit/a-new-article"
                        >
                          <use
                            xlinkHref="icons.svg#edit"
                          />
                        </Link>,
                        "id": "edit",
                        "title": "Edit this article",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": <use
                            xlinkHref="icons.svg#edit"
                          />,
                          "className": "link",
                          "replace": false,
                          "to": "/articles/edit/a-new-article",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "xlinkHref": "icons.svg#edit",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "use",
                        },
                        "type": [Function],
                      },
                      "type": "svg",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <use
                      xlinkHref="icons.svg#flag"
                    />,
                    "id": "flag",
                    "title": "Report this article",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "xlinkHref": "icons.svg#flag",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "use",
                  },
                  "type": "svg",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <use
                      xlinkHref="icons.svg#dislike"
                    />,
                    "className": "",
                    "id": "dislike",
                    "onClick": [Function],
                    "title": "Dislike this article",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "xlinkHref": "icons.svg#dislike",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "use",
                  },
                  "type": "svg",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  5
                </p>,
                "id": "like-count",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 5,
                },
                "ref": null,
                "rendered": 5,
                "type": "p",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <svg
                  className="icon"
                >
                  <use
                    xlinkHref="icons.svg#heart"
                  />
                </svg>,
                "className": "centralizer liked",
                "id": "love",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <use
                    xlinkHref="icons.svg#heart"
                  />,
                  "className": "icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "xlinkHref": "icons.svg#heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "use",
                },
                "type": "svg",
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
